core/Common.php

```diff
@@ -982,10 +982,6 @@ public static function extractLanguageAndRegionCodeFromBrowserLanguage($browserL
                 if (in_array($langIso639 . $regionIso3166, $validLanguages)) {
```

Comment:
Some of the `testExtractLanguageAndRegionCodeFromBrowserLanguage` tests are currently failing because the test data doesn't include the language.region in the valid language array, so the method is unable on this line.

For example, data from the first test is:
```
// browser language, valid languages, expected
['fr-ca',            ['fr'],          'fr-ca'],
```
The test should return 'fr-ca' but is unable to because this is not in the valid languages array, so it returns 'fr' instead and the test fails.

The test data could be updated to `['fr-ca', ['fr', 'fr-ca'], 'fr-ca']` to ensure it choose the region version correctly.

Reply:
Actually the use cases of the method `extractLanguageAndRegionCodeFromBrowserLanguage` are a bit weird.
It accepts a `validLanguages` parameters, but still might return any of the provided languages with an additional region.
I've now changed that behaviour. If valid languages are provided only a code can be returned that exactly matches one of the entries. Otherwise it returns a code if the language is in the list of all valid languages, possibly extended by the region.